src/cgame/cg_weapons.c

```diff
@@ -3601,6 +3601,33 @@ void CG_AddPlayerWeapon(refEntity_t *parent, playerState_t *ps, centity_t *cent)
 		CG_PositionRotatedEntityOnTag(&flash, &gun, "tag_flash");
 	}
 
+	// @XXX hardcode offset flash positions, to correct wrong official
+	// muzzle locations in some cases
+	{
+		vec3_t  forward;
+		float flash_offset = 0.0f;
+
+		switch (weaponNum) {
+			case WP_FLAMETHROWER:
+				// adjust flamethrower muzzle according to cg_fov
+				flash_offset = (((cg.refdef.fov_y / 73.739784 /*90 fov*/) - 1.0) * -3) - 0.4f;
+
+				// pull the flame back slightly while raising
+				if ((cg.snap->ps.weapAnim & ~ANIM_TOGGLEBIT) == WEAP_RAISE) {
+					flash_offset -= 0.5;
+				}
+				break;
+			default:
+				break;
+		}
+
+      if (flash_offset != 0.0f) {
+         AxisToAngles(flash.axis, angles);
+         AngleVectors(angles, forward, NULL, NULL);
+         VectorMA(flash.origin, flash_offset, forward, flash.origin);
+      }
+	}
+
```

Comment:
Don't do a switch. Flamethrower is certainly the only weapon where the tag_flash is miss-placed compared to all other weapons.

Reply:
> Don't do a switch. Flamethrower is certainly the only weapon where the tag_flash is miss-placed compared to all other weapons.
I have 4 more upcoming in my stash tho for engi+covi rifle, but I agree that these should just be modified when I add them.

Reply:
Fine, let keep stuff like this